tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/OrderItemValidatorTests.cs

```diff
@@ -21,5 +29,151 @@ public static void Constructors_VerifyGuardClauses()
 
             assertion.Verify(constructors);
         }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Constructor_NullValidationSettings_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() => _ = new OrderItemValidator(null));
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Validate_NullCommencementDate_ThrowsArgumentNullException(
+            Order order,
+            CreateOrderItemModel model,
+            CatalogueItemType itemType,
+            OrderItemValidator orderItemValidator)
+        {
+            order.CommencementDate = null;
+
+            Assert.Throws<ArgumentNullException>(() => _ = orderItemValidator.Validate(order, model, itemType));
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Validate_AssociatedService_SuccessIsTrue(
+            Order order,
+            CreateOrderItemModel model,
+            OrderItemValidator orderItemValidator)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+            order.CommencementDate.Should().NotBeNull();
+
+            var result = orderItemValidator.Validate(order, model, CatalogueItemType.AssociatedService);
+
+            result.FailedValidations.Should().BeEmpty();
+            result.Success.Should().BeTrue();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Validate_NullDeliveryDate_SuccessIsFalse(
+            Order order,
+            CreateOrderItemModel model,
+            OrderItemValidator orderItemValidator)
+        {
+            model.ServiceRecipients.Should().NotBeNullOrEmpty();
+            order.CommencementDate.Should().NotBeNull();
+
+            var serviceRecipients = model.ServiceRecipients.Select(r =>
```

Comment:
As `r` isn't used we can use a discard here.

Reply:
removed `r`